### ISSUE TITLE
python2 support

### ISSUE THREAD

**Issue by ariebovenberg** [ariebovenberg/snug#11]
blockers for python2 support:
1. returning values from generators with `return` is not possible on python2. A hacky solution would have to be devised, such as raising and catching custom exceptions. Simply raising `StopIteration` is not possible because this breaks on python 3.5+ (PEP479).
2. generator composition is more difficult under python2 without `yield from`.
3. type annotations would have to be removed
4. async components would have to be moved into a separate module, to be imported only on python3.4+

edit: numbered items

**Comment by meowklaski** [comment]
...don't do it! Your code is so clean! Please!

**Comment by ariebovenberg** [comment]
@meowklaski my thoughts as well :)

Although I have been thinking that I might take a stab at points 1 and 2 in the [gentools(https://github.com/ariebovenberg/gentools) repo. 

```python
from gentools import py2_compat

@py2_compat
def repo(owner, name):
    

```

**Comment by meowklaski** [comment]
Taking on points 1 and 2 in that way makes a lot of sense.

**Comment by ariebovenberg** [comment]
implemented in v1.1. closed.
